test/Microsoft.AspNet.Mvc.FunctionalTests/Logging/LoggingTests.cs

```diff
@@ -0,0 +1,199 @@
+                if (string.Equals(typeof(HomeController).AssemblyQualifiedName, isController.Type.ToString()))
+                {
+                    Assert.Equal(
+                        ControllerStatus.IsController,
+                        Enum.Parse(typeof(ControllerStatus), isController.Status.ToString()));
+                }
+                else
+                {
+                    Assert.NotEqual(ControllerStatus.IsController,
+                        Enum.Parse(typeof(ControllerStatus), isController.Status.ToString()));
+                }
+            }
+        }
+
+        [Fact]
+        public async Task ControllerModelValues_LoggedAtStartup()
+        {
+            // Arrange and Act
+            var logs = await GetLogsForStartupAsync();
+            logs = logs.Where(c => c.StateType.Equals(typeof(ControllerModelValues)));
+
+            // Assert
+            Assert.Single(logs);
+            dynamic controller = logs.First().State;
+            Assert.Equal("Home", controller.ControllerName.ToString());
+            Assert.Equal(typeof(HomeController).AssemblyQualifiedName, controller.ControllerType.ToString());
+            Assert.Equal("Index", controller.Actions[0].ActionName.ToString());
+            Assert.Empty(controller.ApiExplorer.IsVisible);
+            Assert.Empty(controller.ApiExplorer.GroupName.ToString());
+            Assert.Empty(controller.Attributes);
+            Assert.Empty(controller.Filters);
+            Assert.Empty(controller.ActionConstraints);
+            Assert.Empty(controller.RouteConstraints);
+            Assert.Empty(controller.AttributeRoutes);
+        }
+
+        [Fact]
+        public async Task ActionDescriptorValues_LoggedAtStartup()
+        {
+            // Arrange and Act
+            var logs = await GetLogsForStartupAsync();
+            logs = logs.Where(c => c.StateType.Equals(typeof(ActionDescriptorValues)));
+
+            // Assert
+            Assert.Single(logs);
+            dynamic action = logs.First().State;
+            Assert.Equal("Index", action.Name.ToString());
+            Assert.Empty(action.Parameters);
+            Assert.Empty(action.FilterDescriptors);
+            Assert.Equal("action", action.RouteConstraints[0].RouteKey.ToString());
+            Assert.Equal("controller", action.RouteConstraints[1].RouteKey.ToString());
+            Assert.Empty(action.RouteValueDefaults);
+            Assert.Empty(action.ActionConstraints.ToString());
+            Assert.Empty(action.HttpMethods.ToString());
+            Assert.Empty(action.Properties);
+            Assert.Equal("Home", action.ControllerName.ToString());
+        }
+
+        [Fact]
+        public async Task Successful_ActionSelection_Logged()
+        {
+            // Arrange
+            var server = TestServer.Create(_services, _app);
+            var client = server.CreateClient();
+            var requestTraceId = Guid.NewGuid().ToString();
+
+            // Act and Assert
```

Comment:
This is definitely not act and assert. The assert is the part where the asserts are :laughing: